src/rgw/rgw_admin.cc

```diff
@@ -6910,7 +6910,7 @@ int main(int argc, const char **argv)
       if (ret < 0) {
         return -ret;
       }
-      if (!role->validate_max_session_duration(dpp())) {
+      if (!role->validate_max_session_duration(dpp(), std::stoull(max_session_duration))) {
```

Comment:
The correct way to validate the value of max session duration here is to reverse the order of calls, like the following:
```
role->update_max_session_duration(max_session_duration);
if (!role->validate_max_session_duration(dpp())) {
ret = -EINVAL;
        return ret;
      }

```

Reply:
Thank you @pritha-srivastava for the clarification. We had the wrong impression that the knob can be used to increase higher session-durations than 43200.

I've added the new evidence. It now addresses 2 things:

- don't allow "role update" to go over the limit.
- let "role create" honor the max-session-duration option.

Also, revised the `long_desc` of the knob to provide some more clarification.

Please have a look.